src/main/java/com/redhat/devtools/lsp4ij/client/features/LSPFormattingFeature.java

```diff
@@ -134,4 +134,18 @@ public void setServerCapabilities(@Nullable ServerCapabilities serverCapabilitie
             rangeFormattingCapabilityRegistry.setServerCapabilities(serverCapabilities);
         }
     }
+
+    // Client configuration settings
+
+    /**
+     * Whether or not server-side on-type formatting is enabled if <code>textDocument/onTypeFormatting</code> is
+     * supported by the language server. Defaults to true.
+     *
+     * @param file the file
+     * @return true if server-side on-type formatting should be enabled for the file; otherwise false
+     */
+    public boolean isTextDocumentOnTypeFormattingEnabled(@NotNull PsiFile file) {

```

Comment:
Added client config to the formatting feature with only this feature flag to start.